backend/gameEndpoint.ml

```diff
@@ -89,56 +91,89 @@ module Make
         let* _ = Firestore.Chat.delete ~request ~id:game_id in
         Dream.empty `OK
 
+    let end_game_elo_update = fun ~(request : Dream.request)  ~(game : Domain.Game.t) ~(winner_enum : Elo.Winner.t): unit Lwt.t ->
+        let type_game : string = game.type_game in
+        let* player_zero_info : Domain.User.EloInfo.t = Firestore.User.get_elo ~request ~user_id:game.player_zero.id ~type_game in
+        let game_player_one : MinimalUser.t = Option.get game.player_one in
+        let game_player_one_id : string = game_player_one.id in
+        let* player_one_info : Domain.User.EloInfo.t = Firestore.User.get_elo ~request ~user_id:game_player_one_id ~type_game in
+        let elo_entry : Elo.EloEntry.t = {
+            elo_info_pair = { player_zero_info; player_one_info };
+            winner = winner_enum;
+        } in
+        let new_elos : Elo.EloInfoPair.t = Elo.CalculationService.new_elos elo_entry in
+        let new_elo_zero : Domain.User.EloInfo.t = new_elos.player_zero_info in
+        let new_elo_one : Domain.User.EloInfo.t = new_elos.player_one_info in
+        let* _ = Firestore.User.update_elo ~request ~user_id:game.player_zero.id ~type_game:game.type_game ~new_elo:new_elo_zero in
+        let* _ = Firestore.User.update_elo ~request ~user_id:game_player_one_id ~type_game:game.type_game ~new_elo:new_elo_one in
+        Lwt.return ()
+
+    let end_game_elo_update_win = fun ~(request : Dream.request) ~(game : Domain.Game.t) ~(winner : MinimalUser.t) : unit Lwt.t ->
+        let winner_enum : Elo.Winner.t =
+            if winner = game.player_zero then Player Zero
+            else Player One in
+        end_game_elo_update ~request ~game ~winner_enum
+
+    let end_game_elo_update_draw = fun ~(request : Dream.request) ~(game : Domain.Game.t): unit Lwt.t ->
+        let winner_enum : Elo.Winner.t = Draw in
+        end_game_elo_update ~request ~game ~winner_enum
+
     (** Resign from a game. Perform 1 read and 2 writes. *)
```

Comment:
repasser pour adapter le nombre d'écriture / lecture

Reply:
in their dad !